### PR TITLE
Moves EditorHeader component to component 'editor' folder

### LINKED ISSUE #145: Move EditorHeader.jsx into subfolder editor/
- Move EditorHeader.jsx and EditorHeader.test.js to editor subfolder and change the name of the file to header.
- dont forget to change the path location of imports as well.

### PR DESCRIPTION
**Pull request by jgreben**
Fixes #145 

The cool thing was that WebStorm made all the changes for me. All I had to do was drop and drag the file...

### CODE REVIEW

**Comment by SaravShah** [comment]
do we want the name to still be editorheader if its in the editor sub folder?

**Comment by jgreben** [comment]
right @SaravShah ...forgot that bit..

**Comment by jgreben** [comment]
WebStorm wasn't as good about refactoring the the imports and component names... :(

**Comment by ndushay** [comment]
oh no!  this PR needs rebasing!

**Comment by ndushay** [comment]
what?  I was trying to rebase!

**Comment by ndushay** [comment]
okay - this is actually Josh's work -- I redid it from the diffs displayed in github.  I think it is NEW and IMPROVED as Josh forgot to rename the component inside the editor/Header.jsx file and also the export.  Whee!

File: src/components/editor/Header.jsx [hunk 1]
```diff
@@ -1,10 +1,8 @@
 // Copyright 2018 Stanford University see Apache2.txt for license
 
 import React, { Component } from 'react'
-import SinopiaLogo from '../styles/editorsinopialogo.png'
-import { Link } from 'react-router-dom'
```
Comment (ndushay) [inline src/components/editor/Header.jsx]: this imports aren't used.  I found this out as I worked on configuring eslint to do accessibility testing with jsx-a11y eslint plugin.

**Pull request merged**